### ISSUE TITLE
Error NaN if scrollbar is used with opacity

### ISSUE THREAD

**Issue by lticadler** [amcharts/amcharts4#2096]
Unfortunately i wasn't able to reproduce the error so easily in a code pen. But hopefully you could detect the error (or tell what i'm doing wrong) with the given information...
Version: 8.6.4

I use a date based linechart (only one series) with a XYChartScrollbar.

The scrollbar series is configured:
```
const scrollSeries1 = me._chart.scrollbarX.scrollbarChart.series.getIndex(0);
scrollSeries1.filters.clear();
scrollSeries1.bullets.clear();
scrollSeries1.fillOpacity = 0.3; // This causes the problem
```
The fillOpacity property causes the problem (don't know which other properties are connected with). If i comment that line out the error later described doesn't happen (as well if i use no scrollbar).

The series is connect = false (but changing it doesn't matter)

When the chart is first time rendered everything is ok.

But when dynamically adding data the error occures.
(The chart is rendered, but a console error happens)
```
dateAxis.min = min; // new min timestamp - the stack below starts here (but only after adding chart data)
dateAxis.max = max; // new max timestmap

chart.invalidateRawData();
chart.addData(someNewData, removeCount);
```
(no data of mine has NaN or undefined values but they are not equidistant)

Stack from console:
```
Error: <path> attribute d: Expected number, "… L0,-100000 L0,NaN L0,NaN L0,…".
(anonymous function)              	AMElement.ts:680
(anonymous function)              	Object.ts:120
l                                 	Object.ts:105
Qkdp.e.each                       	Object.ts:119
t.attr                            	AMElement.ts:671
e.setPath                         	Sprite.ts:7744
set                               	Sprite.ts:7728
e.drawSegment                     	LineSeriesSegment.ts:151
e.drawSegment                     	LineSeries.ts:657
e.closeSegment                    	LineSeries.ts:636
e.openSegment                     	LineSeries.ts:575
e.openSegmentWrapper              	LineSeries.ts:490
e.validate                        	LineSeries.ts:405
(anonymous function)              	ValueAxis.ts:1344
e.each                            	Array.ts:95
t.each                            	List.ts:905
e.getMinMax                       	ValueAxis.ts:1343
e.handleExtremesChange            	ValueAxis.ts:2060
e.handleExtremesChange            	DateAxis.ts:2237
(anonymous function)              	EventDispatcher.ts:444
(anonymous function)              	EventDispatcher.ts:324
e.each                            	Array.ts:95
t._eachListener                   	EventDispatcher.ts:294
t.dispatchImmediately             	EventDispatcher.ts:322
e.dispatchImmediately             	Base.ts:1335
e.processValues                   	XYSeries.ts:1693
e.validateDataItems               	Series.ts:616
e.validateDataItems               	XYSeries.ts:1333
t.update                          	System.ts:218
(anonymous function)              	System.ts:405
Async call from requestAnimationFrame
u                                 	AsyncPending.ts:40
t.requestFrame                    	System.ts:404
e.invalidateDataRange             	Component.ts:969
e.invalidateProcessedData         	Component.ts:2092
e.handleDataItemWorkingValueChange	Series.ts:1065
e.handleDataItemWorkingValueChange	XYSeries.ts:2477
e.setWorkingValue                 	DataItem.ts:832
e.setValue                        	DataItem.ts:748
e.setDate                         	DataItem.ts:950
set                               	XYSeries.ts:178
(anonymous function)              	Component.ts:713
(anonymous function)              	Object.ts:120
l                                 	Object.ts:105
Qkdp.e.each                       	Object.ts:119
e.processDataItem                 	Component.ts:674
e.processDataItem                 	XYSeries.ts:1065
(anonymous function)              	Component.ts:1186
e.each                            	Array.ts:95
t.each                            	List.ts:905
o                                 	Component.ts:1183
e.validateData                    	Component.ts:1176
e.validateData                    	XYChart.ts:614
t.update                          	System.ts:157
(anonymous function)              	System.ts:405
Async call from requestAnimationFrame
u                                 	AsyncPending.ts:40
t.requestFrame                    	System.ts:404
e.invalidate                      	Sprite.ts:1165
e.invalidate                      	Container.ts:482
set                               	ValueAxis.ts:1587
```

**Comment by zeroin** [comment]
Thank you for reporting. How many data items do you have on your series when this happens?

**Comment by lticadler** [comment]
> Thank you for reporting. How many data items do you have on your series when this happens?

When i tested there were 30 to 35 items. The distance between them is in the range of 30 to 60 seconds.

**Comment by zeroin** [comment]
I've tried various ways to reproduce this but with no luck. So I must say I am stuck here, unless you could reproduce it on codepen...

**Comment by lticadler** [comment]
I was able to create a pen for that problem. Seems to be a Chrome only problem. Didn't get in in Firefox.
Attention - the error does not show up in codepen console. You have to open chrome dev tools. Best to open dev tools before you enter the url.

The setTimeout at the end triggers the error.
If line 26 (series.fillOpacity = 0.3;) is removed no error happens.

https://codepen.io/chrisadler/pen/PoqojVj

![image](https://user-images.githubusercontent.com/58069494/73939008-cc92b280-48e8-11ea-9412-9b34b82266f5.png)

**Comment by zeroin** [comment]
Thanks a lot, I am looking into it and will let you know as soon as this is fixed.

**Comment by zeroin** [comment]
IT should work fine with v 4.8.9

**Comment by lticadler** [comment]
Thanks :) will test it on monday

**Comment by Christian-Adler** [comment]
No more errors :)

**Comment by lticadler** [comment]
No more errors :)

**Comment by lticadler** [comment]
In the pen now there are no more errors...

Unfortunately in my productive app there are still the same errors after downloading version 4.9.0.
The stack has the same files with a few new line numbers.

I'll try to get a new pen to get these errors...

New Stack:
```
Error: <path> attribute d: Expected number, "…00 C0,-100000 0,NaN 0,-100000 C0…".
(anonymous function)              	AMElement.ts:680
(anonymous function)              	Object.ts:120
l                                 	Object.ts:105
Qkdp.e.each                       	Object.ts:119
t.attr                            	AMElement.ts:671
e.setPath                         	Sprite.ts:7760
set                               	Sprite.ts:7744
e.drawSegment                     	LineSeriesSegment.ts:144
e.drawSegment                     	LineSeries.ts:657
e.closeSegment                    	LineSeries.ts:636
e.openSegment                     	LineSeries.ts:575
e.openSegmentWrapper              	LineSeries.ts:490
e.validate                        	LineSeries.ts:405
(anonymous function)              	ValueAxis.ts:1344
e.each                            	Array.ts:95
t.each                            	List.ts:905
e.getMinMax                       	ValueAxis.ts:1343
e.handleExtremesChange            	ValueAxis.ts:2060
e.handleExtremesChange            	DateAxis.ts:2237
(anonymous function)              	EventDispatcher.ts:444
(anonymous function)              	EventDispatcher.ts:324
e.each                            	Array.ts:95
t._eachListener                   	EventDispatcher.ts:294
t.dispatchImmediately             	EventDispatcher.ts:322
e.dispatchImmediately             	Base.ts:1380
e.processValues                   	XYSeries.ts:1694
e.validateDataItems               	Series.ts:616
e.validateDataItems               	XYSeries.ts:1334
t.update                          	System.ts:218
(anonymous function)              	System.ts:405
Async call from requestAnimationFrame
u                                 	AsyncPending.ts:40
t.requestFrame                    	System.ts:404
e.invalidateDataRange             	Component.ts:969
e.invalidateProcessedData         	Component.ts:2092
e.handleDataItemWorkingValueChange	Series.ts:1065
e.handleDataItemWorkingValueChange	XYSeries.ts:2476
e.setWorkingValue                 	DataItem.ts:832
e.setValue                        	DataItem.ts:748
e.setDate                         	DataItem.ts:950
set                               	XYSeries.ts:178
(anonymous function)              	Component.ts:713
(anonymous function)              	Object.ts:120
l                                 	Object.ts:105
Qkdp.e.each                       	Object.ts:119
e.processDataItem                 	Component.ts:674
e.processDataItem                 	XYSeries.ts:1066
(anonymous function)              	Component.ts:1186
e.each                            	Array.ts:95
t.each                            	List.ts:905
o                                 	Component.ts:1183
e.validateData                    	Component.ts:1176
e.validateData                    	XYChart.ts:614
t.update                          	System.ts:157
(anonymous function)              	System.ts:405
Async call from requestAnimationFrame
u                                 	AsyncPending.ts:40
t.requestFrame                    	System.ts:404
e.invalidate                      	Sprite.ts:1165
e.invalidate                      	Container.ts:482
set                               	ValueAxis.ts:1587
```

**Comment by lticadler** [comment]
Don't know what i changed - now the error is gone...
It shouldn't be a cache thing - the different stack proves that is was the new version... confused.

I'll keep an eye on it. But for now i think this issue could be closed...
Thanks again.

**Comment by lticadler** [comment]
Still happens sporadically - can't get it exactly when. But if it is in this error "state" it happens on every new data.

Sometimes i see an additional line in the scrollbar (seems to connect the start and end of the svg path). Don't know it's on coincidence.
![image](https://user-images.githubusercontent.com/58069494/74307667-481bb600-4d66-11ea-8126-83e373473aae.png)

**Comment by zeroin** [comment]
Thanks - this helped, I think I might know what to do to solve.

**Comment by lticadler** [comment]
Got it nearly reproducible - happens after multiple data updates. This why maybe a timing problem...

Changed the pen to update the data not once but in interval. After 10 to 40 iterations i always get the error in the chrome console.
https://codepen.io/chrisadler/pen/PoqojVj

![image](https://user-images.githubusercontent.com/58069494/74310641-ebbc9480-4d6d-11ea-9542-fa8435f50f79.png)

**Comment by martynasma** [comment]
Fixed in [4.9.1](https://github.com/amcharts/amcharts4/releases/tag/4.9.1).

## [4.9.1] - 2020-02-14

### Added
- `precision` added to `MapLine` (default `0.1`). If line has `shortestDistance = true` set, it will be bent according to to projection. If `precision` is larger than the distance (degrees) between line's end points, no such bending will occur. Set it to large number for perfectly straight lines.
- `precision` added to `MapPolygon` (default `0.5`). Polygon's side lines are bent according to to projection. If `precision` is larger than the distance (degrees) between side's end points, no such bending will occur. Set it to large number for perfectly straight lines.

### Fixed
- JSON config: referring to a `parent` by id is now not dependent on order of elements.
- Series elements (e.g. columns) were not dispatching "over" event when `XYCursor` was being used.
- When `Legend` was triggering hover over `PieSeries` slice (or other `PercentSeries`), the slice did not dispatch `"over"` event.
- Hidden `ForceDirectedNode` (if hidden using API) was not showing if legend item was clicked to show it again.
- `path` adapter was not working on `Sprite`.
- Adding `data` to a `XYChart` with an `XYChartScrollbar` with a series was sometimes resulting JS error (if some data was missing).
- Adding `AxisBreak` to `CategoryAxis` could result JS error.
- `groupFields` were not copied to series in `XYChartScrollbar`, or any other cloned series.
- If a bullet was already disabled when `disabled` property field was set, that bullet did not update correctly after data change.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
